### DISCUSSION TITLE
Reading small files is very slow

### DISCUSSION THREAD

**Discussion by dengdalue** [minio/minio#17131] [Q&A]
minio cluster：
Four VMS. Each VM has one SSD

minio version：
minio version RELEASE.2023-03-22T06-36-24Z (commit-id=035791669eb43f90bc38cd68f3dc87f65d622993)
Runtime: go1.19.7 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.

OS version：
CentOS Linux release 7.9.2009 (Core)

Config：
#!/bin/env bash
export MINIO_ROOT_USER=xxx
export MINIO_ROOT_PASSWORD=xxx
export MINIO_PROMETHEUS_AUTH_TYPE=public
minio server --config-dir /file/minio/etc --console-address :9001 http://192.168.77.165/file1 http://192.168.77.166/file1 http://192.168.77.167/file1 http://192.168.77.168/file1

Run the following command to mount bucket test to the /file/test directory：
s3fs -o uid=1001,gid=1001 -o url=http://192.168.77.165:9000 -o allow_other -o nonempty -o no_check_certificate -o use_path_request_style -o umask=000 test /file/test

Copying small files from /file/test to other directories is very slow

**Comment by klauspost** [comment]
We do not offer performance reviews as part of our community support. Also using s3fs is highly suboptimal and will likely yield both slow speed and inconsistencies.

S3 is not a filesystem.

**Comment by AlexZIX** [comment]
As my opinion s3fs is a really slow-speed FS. I've also tried GeeseFS for mount buckets but it's very unstable when more that 100K files in your bucket. For now I use the rclone if I need to mount bucket into linux as filesystem. Suggest you to try it.
